code/controllers/subsystem/nightshift.dm

```diff
@@ -0,0 +1,59 @@
+SUBSYSTEM_DEF(nightshift)
+	name = "Night Shift"
+	init_order = INIT_ORDER_NIGHTSHIFT
+	priority = FIRE_PRIORITY_NIGHTSHIFT
+	wait = 600
+	flags = SS_NO_TICK_CHECK
+
+	var/nightshift_active = FALSE
+	var/nightshift_start_time = 702000		//7:30 PM, station time
+	var/nightshift_end_time = 270000		//7:30 AM, station time
+	var/nightshift_first_check = 30 SECONDS
+
+	var/high_security_mode = FALSE
+
+/datum/controller/subsystem/nightshift/Initialize()
+	if(!config.enable_night_shifts)
+		can_fire = FALSE
+	if(config.randomize_shift_time)
```

Comment:
Did you put this here because of the ticker not being updated yet?

Reply:
Correct.